dev/com.ibm.ws.cdi.ejb_fat/fat/src/com/ibm/ws/cdi/ejb/apps/newTimer/EjbSessionBean.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.cdi.ejb.apps.newtimer;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.annotation.Resource;
+import javax.ejb.EJB;
+import javax.ejb.Local;
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+import javax.ejb.Timeout;
+import javax.ejb.Timer;
+import javax.ejb.TimerConfig;
+import javax.ejb.TimerService;
+import javax.inject.Inject;
+
+import javax.enterprise.context.ContextNotActiveException;
+
+/**
+ * Session Bean implementation class EjbSessionBean
+ */
+@Stateless
+@Local(EjbSessionBeanLocal.class)
+@LocalBean
+@MyCDIInterceptorBinding
+public class EjbSessionBean implements EjbSessionBeanLocal {
+    private static final long TIMEOUT = 1;
+
+    @Resource
+    TimerService timerService;
+
+    @Inject RequestScopedBean rsb;
+
+    @Inject SessionScopedBean ssb;
+
+    private int requestScopedIndex = -1;
+    public static Boolean canAccessRequestScope = null;
+    public static Boolean seperateRequestScopes = null;
+    public static Boolean sessionScopeInactive = null;
+
+
+    @Timeout
+    void ping(Timer timer) {
+        //Test One. Is a request scope active in the context of an EJB timer?
+        if (rsb.toString() != null) {
+            canAccessRequestScope = true;
+        } else {
+            canAccessRequestScope = false;
+        }
+
+        //Test two. Is it a different request scope to outside the timer?
+        if (rsb.getIndex() != requestScopedIndex) {
+            seperateRequestScopes = true;
+        } else {
+            seperateRequestScopes = false;
+        }
```

Comment:
Given that your first test is checking the request scope is active, it should really be in a try-catch for `ContextNotActiveException` like the session scope test (even though we do expect it to be active).

The second test also relies on the request scope being active, so it should also be within a try-catch.

If you really wanted to get fancy, you could make sure that any exception thrown is passed back out and thrown from the test method as well so that you get the actual `ContextNotActiveException` with the right line number in the test result.